clustering/ejb/infinispan/src/main/java/org/wildfly/clustering/ejb/infinispan/BeanExpirationScheduler.java

```diff
@@ -87,51 +88,59 @@ public void cancel(Locality locality) {
 
     @Override
     public void close() {
-        for (Future<?> future: this.expirationFutures.values()) {
-            future.cancel(false);
+        final Future<?> task;
+        synchronized (this) {
+            if (expireTask == null) {
+                return;
+            }
+            task = expireTask;
+            expireTask = null;
```

Comment:
Should we be concerned that the expireTask can be reset after this synchronized block completes?  Whose cancels the Future in that case?

Reply:
I agree, I fixed that by checking if expireTask is different from null at the end of the file, before trying to reschedule it and reset the field.